peer/peer.go

```diff
@@ -39,6 +41,29 @@ type Peer struct {
 	AuthInfo credentials.AuthInfo
 }
 
+// String ensures the Peer types implements the Stringer interface in order to
+// allow to print a context with a peerKey value effectively.
+func (p *Peer) String() string {
+	if p == nil {
+		return "Peer<nil>"
+	}
+	sb := &strings.Builder{}
+	sb.WriteString("Peer{")
+	if p.Addr != nil {
```

Comment:
mm, wouldnt the peer always have the peer's Addr?

Reply:
The documentation didn't really said and I wanted to make sure to avoid pathological cases such as the above one, where someone creates an empty `Peer` and calls `String` on it.

Reply:
I agree, the stringer should handle all the "pathological cases". For eg, what if this is called from a unit test without Addr set.